woocommerce/woocommerce-ios
# Dynamic Dashboard: Reviews Card Reload Functionality Part 2

Part of: #12742


## Description


This PR adds more functionalities to the reviews card:
1. Filtering reviews by All / Approved / Hold (includes updating `ProductReviewAction.synchronizeProductReviews` to accept a status parameter)
2. Displaying read/unread state of the reviews (shown in the bubble icon color), by adding notification fetching.
3. Tapping individual review in the card to open its details screen.
4. Moves logic for "View all reviews" button navigation to View Controller.

This PR does not include these, they will be added in a separate PR:
1. Proper empty/loading/error states.
2. unit tests
3. Tracking

## Testing instructions


1. Start with a WordPress.com test site with a variety of review statuses (specifically approved and hold).
2. Go to the test site and write a new review,
3. Start app,
5. Check Reviews card (enable first in Dashboard Edit screen if you haven't),
6. Ensure that the new review added on step 2 is shown with blue speech bubble icon,
7. Ensure that the other reviews are shown in blue or gray bubble icon, depending if they have been read or not.
8. Tap a review, ensure this open the review's details,
9. Go back to Dashboard,
10. Test the All / Approved / Hold filtering (please ignore the loading visual for now). Ensure that the shown reviews are correct, and that the label next to "Status" on the Dashboard shows the correct filtering.
11. Test the "View all reviews" button and ensure it still works well.

## Video:

https://github.com/woocommerce/woocommerce-ios/assets/266376/2244ac33-fd1b-4d0c-babd-875bd76a503e
